src/psyclone/tests/psyad/tl2ad_test.py

```diff
@@ -455,21 +562,39 @@ def test_generate_adjoint_errors():
 
 @pytest.mark.xfail(reason="issue #1235: caplog returns an empty string in "
                    "github actions.", strict=False)
-def test_generate_adjoint_logging(caplog, fortran_reader, fortran_writer):
+def test_generate_adjoint_logging(
+        caplog, fortran_reader, fortran_writer, tmpdir):
     '''Test that logging works as expected in the generate_adjoint()
     function.
 
     '''
     tl_fortran_str = (
-        "program test\n"
-        "real :: a\n"
-        "a = 0.0\n"
-        "end program test\n")
+        "module test_mod\n"
+        "contains\n"
+        "  subroutine test1(a)\n"
+        "  real :: a\n"
+        "  a = 0.0\n"
+        "  end subroutine\n"
+        "  subroutine test2(a)\n"
+        "  real :: a\n"
+        "  a = 0.0\n"
+        " end subroutine\n"
+        "end module\n")
     expected_ad_fortran_str = (
-        "program test_adj\n"
-        "  real :: a\n\n"
-        "  a = 0.0\n\n"
-        "end program test_adj\n")
+        "module adj_test_mod\n"
+        "  implicit none\n"
+        "  public\n\n"
+        "  public :: adj_test1, adj_test2\n\n"
```

Comment:
I don't think this is going to be correct anymore (I recently improved the backend so that it no longer outputs separate visibility statements if the symbols have the default visibility). That in turn shows that we need to tighten up the use of xfail in this test so that it checks the caplog bit explicitly.

Reply:
Yes well spotted. I've updated this test and the others in this file that used a pytest.xfail decorator.

Reply:
Great - that's much better now.